pkg/asset/agent/manifests/nmstateconfig_test.go

```diff
@@ -41,6 +82,16 @@ func TestNMStateConfig_Generate(t *testing.T) {
 				assert.Equal(t, tc.expectedError, err.Error())
 			} else {
 				assert.NoError(t, err)
+				assert.Equal(t, tc.expectedConfig, asset.Config)
+				assert.NotEmpty(t, asset.Files())
+
+				configFile := asset.Files()[0]
+				assert.Equal(t, "cluster-manifests/nmstateconfig.yaml", configFile.Filename)
+
+				var actualConfig []*aiv1beta1.NMStateConfig
+				err = yaml.Unmarshal(configFile.Data, &actualConfig)
```

Comment:
I think you'll have to use `strings.Split(configFile.Data, "\n---\n")` to split these.

Alternatively, here is the code we're actually using to read the file in the NMStateConfig asset:
https://github.com/openshift/installer/blob/agent-installer/pkg/asset/agent/manifests/nmstateconfig.go#L217-L245